src/lib.rs

```diff
@@ -1117,14 +1130,27 @@ impl<T: Counter> Histogram<T> {
 
     /// Computes the matching histogram value for the given histogram bin.
     pub fn value_for(&self, index: usize) -> u64 {
+        // TODO what to do about indexes that are beyond what can be represented by this histogram?
+        // Pretty easy to end up shifting off the high end of a u64, e.g. asking for a big index
+        // when sigfigs is high.
```

Comment:
Once I'm done going over this (which might produce another `TODO` or two) I'll look into retroactively fixing those issues separately if it can be done tidily without replicating these type changes.